### ISSUE TITLE
Order of Data Augmentation and Normalization

### ISSUE THREAD

**Issue by Separius** [autonomousvision/stylegan_xl#44]
Hi,
first of all, thank you for your clean code; second, I think I found a (probably not that important) bug.
As far as I understand based on the DA paper, one should apply [DiffAugment](https://github.com/autonomousvision/stylegan_xl/blob/main/training/diffaug.py) to images when they are in the (-1, 1) range, I'm saying this based on their codebase, they first scale the input from (0, 255) to (-1, 1) in [here](https://github.com/mit-han-lab/data-efficient-gans/blob/master/DiffAugment-stylegan2-pytorch/training/training_loop.py#L261) and then apply their DA function [here](https://github.com/mit-han-lab/data-efficient-gans/blob/master/DiffAugment-stylegan2-pytorch/training/loss.py#L54)
but in this repository, I think you first scale to (-1, 1) (here)[https://github.com/autonomousvision/stylegan_xl/blob/main/training/training_loop.py#L315] and then scale to (0, 1) (here)[https://github.com/autonomousvision/stylegan_xl/blob/main/pg_modules/discriminator.py#L197] then normalize based on the pretrained feature extractor (which is correct and it should be applied to (0, 1) images) and then finally you apply DA, isn't it more accurate if you first apply DA then move to (0, 1) and then normalize?

**Comment by xl-sr** [comment]
Hey, 
very good point 🤔 it likely doesn't change the results (i played around with augmentation, and usually the end result is not that different). I'll switch it around and test it, thanks!

**Comment by xl-sr** [comment]
Fixed in 890519e061da9a7ac5eb3e012f32d5bf7211e015.

In my quick experiments, the results slightly improved :) Thanks for bringing it to my attention!

**Comment by Separius** [comment]
Damn, that was fast :), glad it helped.
